### DISCUSSION TITLE
Show recently opened files

### DISCUSSION THREAD

**Discussion by BriT999** [sumatrapdfreader/sumatrapdf#3104] [Ideas]
Loving Sumatra - keep up the great work :)

Similar to Word, Excel etc, it would very useful if there was a file menu item that showed most recently opened files (like most recent 15 or so). Is that hard to implement?

**Comment by tactilis** [comment]
It already has it:
 
![image](https://user-images.githubusercontent.com/4060092/202228309-4dd2ddae-4462-4f44-87f4-24f75547e563.png)

**Reply by BriT999** [reply]
this only works if the setting to "Remember open files" is selected which means that each time SumatraPDF is opened, then all recent files are opened as well in addition to this list becoming viewable. My request is slightly different - to be able to only view the list of recently opened files (as shown in your screenshot) without actually having all of these files opened too

I hope i have explained well. Does it make sense and have I missed an existing setting that allows this?

**Reply by GitHubRulesOK** [reply]
SumatraPDF can remember files without sessions thats an advanced setting
however, without sumatraPDF remember then that's a windows feature add SumatraPDF to taskbar and use recents
![image](https://user-images.githubusercontent.com/12754913/202251272-8014243f-fc6f-474b-aa88-7d058d0f9b19.png)

**Comment by tactilis** [comment]
Or as @GitHubRulesOK says, there is an Advanced setting:

```
RememberOpenedFiles = true
RememberStatePerDocument = true
RestoreSession = true
```

Change RestoreSession to false

**Comment by slugsir** [comment]
Somewhat related - is there a way where if you have multiple windows open, hence multiple sessions, Sumatra can remember each session independently? Currently it seems to only remember one of them.

**Reply by GitHubRulesOK** [reply]
for that scenario you would be better served with separate sessions each with its own history, not ideal, but in certain cases desirable. To start sessions independently you would need separate settings folders thus using `SumatraPDF  -appdata "folder" "path/any.pdf"`

**Comment by Darkzeroprojects-coding** [comment]
Is it possible to expand the recent files list up from the default shown?
**If so** how many?
**If not** what be best for one using Sumatra say different books of a specifics like
5 for programming, 5 for Mythology books, and 7 martial arts books?
**Use the method suggested for Slugsir** and setup seperate settings folders?

**Comment by GitHubRulesOK** [comment]
@Darkzeroprojects-coding 
With the introduction of Windows, Microsoft provided visual folders under My Documents, where you can aggregate topics such as below here. The subfolder is Microsoft Learn. (Around 160 pdf files of some interest to me.) Using SumatraPDF search and display I can instantly find 17 with a keyword in any documents sub folder including with preview or read and double click to open.

![image](https://user-images.githubusercontent.com/12754913/204886534-d855169b-ace5-4125-b349-499b33101b6b.png)

Without search I can see all 160
![image](https://user-images.githubusercontent.com/12754913/204888188-7f355e4f-290b-4d4d-939b-9e6635b05181.png)


However, they did not extend that ability to Recent's
![image](https://user-images.githubusercontent.com/12754913/204890678-9d8679be-c757-4eef-942c-731b8be87eb8.png)

Even recent Files can be hit or miss !!
![image](https://user-images.githubusercontent.com/12754913/204894637-692523b0-6527-4594-a4f5-ce6c2d2718b1.png)
